client/ayon_core/pipeline/workfile/utils.py

```diff
@@ -124,3 +142,624 @@ def should_open_workfiles_tool_on_launch(
     if output is None:
         return default_output
     return output
+
+
+def _get_event_context_data(
+    project_name: str,
+    folder_entity: dict[str, Any],
+    task_entity: dict[str, Any],
+    host_name: str,
+):
```

Comment:
In my ideal unicorn world this would also return dataclass. Also for this kind of function where project name, folder entity, task entity is always mandatory, shouldn't we wrap it to dataclass too? It would simplify things.

```suggestion
def _get_event_context_data(
    project_name: str,
    folder_entity: dict[str, Any],
    task_entity: dict[str, Any],
    host_name: str,
) -> dict[str, Any]:
```